iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -176,6 +179,20 @@ public RecordReader<Void, Container<Record>> getRecordReader(InputSplit split, J
     }
   }
 
+  private static void validateFilesWithinTableDirectory(InputSplit split, JobConf job) throws IOException {
+    boolean dataFilesWithingTableLocationOnly =
+        job.getBoolean(HiveConf.ConfVars.HIVE_ICEBERG_ALLOW_DATA_IN_TABLE_LOCATION_ONLY.varname,
+            HiveConf.ConfVars.HIVE_ICEBERG_ALLOW_DATA_IN_TABLE_LOCATION_ONLY.defaultBoolVal);
+    if (dataFilesWithingTableLocationOnly) {
+      Path tableLocation = new Path(job.get(InputFormatConfig.TABLE_LOCATION));
+      for (String location : split.getLocations()) {
```

Comment:
is that reliable?
````
 getLocations() won't be accurate when called on worker nodes and will always return "*"
````

Reply:
check the method code:
````
  public String[] getLocations() {
    // The implementation of getLocations() is only meant to be used during split computation
    // getLocations() won't be accurate when called on worker nodes and will always return "*"
    if (locations == null && conf != null) {
      boolean localityPreferred = conf.getBoolean(InputFormatConfig.LOCALITY, false);
      locations = localityPreferred ? Util.blockLocations(task, conf) : ANYWHERE;
    } else {
      locations = ANYWHERE;
    }

    return locations;
  }
````
when invoked 2nd time it returns "*" always :) nice design

Reply:
That wasn't a cool thing I believe, I changed to use the FileScanTasks, that is what contains the actual files to be read.

I even added a test case where a file is actually readable, so the path would be extracted here & then for normal read, to ensure some only one time read stuff doesn't break the normal reads